File: javapatterns/src/main/java/org/iut/nantes/adapter/chimie/OldDB.java
package org.iut.nantes.adapter.chimie;

import java.util.HashMap;

class OldDB
{
  private HashMap<String, Molecule> database_;

  public OldDB()
  {
    database_ = new HashMap<String, Molecule>();
  }

  public void oldAdd( Molecule m ) { database_.put( m.getName(), m ); }
  public boolean exists( String name ) { return database_.containsKey( name ); }
  public Molecule oldMolecule( String name ) { return database_.get( name ); }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/adapter/chimie/OldDB.kt
package org.iut.nantes.adapter.chimie

import java.util.*

internal open class OldDB {
    private val database_: HashMap<String?, Molecule>
    fun oldAdd(m: Molecule) {
        database_[m.name] = m
    }

    open fun exists(name: String?): Boolean {
        return database_.containsKey(name)
    }

    open fun oldMolecule(name: String?): Molecule? {
        return database_[name]
    }

    init {
        database_ = HashMap()
    }
}